activesupport/lib/active_support/message_encryptor.rb

```diff
@@ -29,6 +33,9 @@ def self.dump(value)
     end
 
     class InvalidMessage < StandardError; end
+    class UnexpectedAlgorithm < StandardError; end
+    class InvalidFormat < StandardError; end
```

Comment:
This backwards compatibility is really confusing. Does this mean that new tokens should behave in exactly the same way as the old tokens? Raise the same exceptions that were raised previously at the same places ?  I think I really need to clear my head on this :sweat: